src/Components/EditInventory/EditInventory.js

```diff
@@ -0,0 +1,192 @@
+import { useEffect, useState } from 'react';
+import { Link, useHistory, useParams } from 'react-router-dom';
+import "./EditInventory.scss";
+import axios from 'axios';
+import errorIcon from '../../Assets/Icons/error-24px.svg'
+
+const EditInventory = () => {
+    const history = useHistory();
+    const params = useParams();
+    const [inStock, setInStock] = useState(true);
+    const [itemDetails, setItemDetails] = useState({
+        itemName: "",
+        warehouseName: "",
+        status: "",
+        quantity: 0,
+        description: "",
+        id: "",
+        warehouseId: "",
+
+    });
+    const [warehouses, setWarehouses] = useState({});
+    const [nameError, setNameError] = useState(false);
+    const [descError, setDescError] = useState(false);
+    const [quantityError, setQuantityError] = useState(false);
+
+    useEffect (() => {
+        axios.get(`http://localhost:8080/inventory/${params.id}`).then(res => {
+            setItemDetails(res.data[0]);
+            updateTheStock(res.data[0]);
+        }).catch(err => {
+            console.log(err);
+        });
+        axios.get(`http://localhost:8080/warehouse/`).then(res => {
+            setWarehouses(res.data);   
+        });
+
+    }, [params]);
+
+    const updateTheStock = details => {
+        if(details.status === "In Stock" && details.quantity > 0) {
+            setInStock(true);
+        }
+        else {
+            setInStock(false);
+        }
+    }
+
+    const handleFormChange = e => {
+
+        if (e.target.name === 'warehouseName') {
+                const wh = warehouses.filter(wh => wh.name === e.target.value);
+                console.log(wh[0].id);
+                setItemDetails({...itemDetails, warehouseId: wh[0].id});
```

Comment:
```suggestion
                const updatedItemDetails = {...itemDetails }
                updatedItemDetails['warehouseID'] = wh[0].id;
                updatedItemDetails['warehouseName'] = e.target.value;
                setItemDetails(updatedItemDetails);
```

Unfortunately the warehouse thing is still happening. I think there's two parts to this. First is the key name should be `warehouseID` instead of `warehouseId` (same as the line 18) thing, and the second is that the `warehouseName` itself depends on the update in line 55, which is a bit confusing. My suggested code here is to make it obvious what's being updated.

Another thing is that for the parent if statement, it would be much clearer if we had something like the following:
```
if (e.target.name === 'warehouseName') {
    ...
} else if (e.target.name === 'status') {
    ....
} else {
    setItemDetails({...itemDetails, [e.target.name]: e.target.value});
}
```

The above code makes it so that we're very clear "we only want to do something special if the field is `warehouseName` or if the field is `status`"